### ISSUE TITLE
Randomly I am getting logged out of all sites

### ISSUE THREAD

**Issue by mark-orion** [brave/brave-browser#39758]
### Description

I am getting logged out of ALL sites on Brave at random times.
After the problem appeared, I tried with a totally new (all user data removed) and clean (no extensions, no sync) install of Brave. This did not solve the problem.
Clearing browser data on exit is not enabled.

### Steps to reproduce

This is difficult to reproduce as I cannot see any reason why these logouts happen. They sometimes happen with brave just running in the background for a while.

### Actual result

All logins are gone. Sometimes but not always even the stored passwords in the password manager disappear. 
This means I repeatedly have to go through the annoying process of logging back into all my favorite sites (including 2FA).

### Expected result

Logins are stored and re-opening sites works without problem

### Reproduces how often

Intermittent issue

### Brave version (brave://version info)

1.67.123 Chromium: 126.0.6478.126 (Official Build) (64-bit) 

### Channel information

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

I am using a nearly fresh install of Ubuntu Studio 24.04. This distro is based on KUbuntu and uses a special low latency kernel for audio production.

**Comment by PirlouitJambon** [comment]
Same problem here, this is a real nigntmare because the lost credentials are my customers credentials and they will think that I'm stupid to lose them. They will not trust me anymore.

Please help us.

**Comment by PirlouitJambon** [comment]
Hello,

Can someone please help us ?
I will lose all my customers if I don't recover the saved passwords.

Thank you

**Comment by harvey1344** [comment]
I have also had this issue, all my saved passwords just dissapeared

**Comment by popy2k14** [comment]
Same issue here. 
Sometimes I am just logged out of ex. GitHub.
I am using Bitwarden self hosted, so Relogon is not an issue.

**Comment by fmarier** [comment]
This might be due to the password store (which is also used to encrypt cookies) not being detected properly on your machine: https://support.brave.com/hc/en-us/articles/29808985123085-Sensitive-data-storage

Try specifying it manually on the command-line to see whether you can find your missing passwords/cookies: https://source.chromium.org/chromium/chromium/src/+/refs/tags/128.0.6613.85:docs/linux/password_storage.md

We may be able to improve our detection logic if we understand more what's happening on your machine, but the first step is to confirm that it's related to the password store.

**Comment by popy2k14** [comment]
Thats also happen on barve mobile for android sometimes!

**Comment by fmarier** [comment]
> Thats also happen on barve mobile for android sometimes!

Do you have Brave Sync enabled?

**Comment by popy2k14** [comment]
Yes, enabled all but password sync is off (using bitwarden).

**Comment by szilardszaloki** [comment]
@popy2k14 Would you be able to run Brave from the terminal with `--enable-logging --v=1` and share with me the resulting `chrome_debug.log` (from your user data directory)?

**Comment by fmarier** [comment]
> it is using kwalletmanager5 in my desktop environment

Which desktop environment?

Does it work fine if you start the browser with [`--password-store=kwallet5`](https://source.chromium.org/chromium/chromium/src/+/refs/tags/128.0.6613.85:docs/linux/password_storage.md)?

**Comment by Nukleari** [comment]
> This might be due to the password store (which is also used to encrypt cookies) not being detected properly on your machine: https://support.brave.com/hc/en-us/articles/29808985123085-Sensitive-data-storage
> 
> Try specifying it manually on the command-line to see whether you can find your missing passwords/cookies: https://source.chromium.org/chromium/chromium/src/+/refs/tags/128.0.6613.85:docs/linux/password_storage.md
> 
> We may be able to improve our detection logic if we understand more what's happening on your machine, but the first step is to confirm that it's related to the password store.

I'm also having this issue, its definitely related to the password store.
I have hyprland and kwallet, I unlock my kwallet via pam [wiki](https://wiki.archlinux.org/title/KDE_Wallet): paragraphs 2.1 and 3.6
I have noticed that when unlocking pam with ```/usr/lib/pam_kwallet_init &``` instead of ```/usr/lib/pam_kwallet_init```,  the issue occurred after boot sometimes.
Now It only happens when my pc locks (using hypridle and hyprlock ```loginctl lock-session``` the strange thing is that I can't tell why exactly, as other stuff doesn't have an issue with it, like network manager can find vpn passwords and stuff.

**Comment by Equinox984** [comment]
I have the exact same problem, since I use Proton Pass I don't have problem with the passwords. But it is very annoying

**Comment by catfluoride** [comment]
@PirlouitJambon 
Slightly OT, but generally speaking, it is better to use a dedicated password manager, instead of storing them in a browser.
Try Bitwarden or ProtonPass, which are free and very reliable.

**Comment by CaseyLabs** [comment]
Just an FYI: I imagine most users will find that this issue is resolved with the `BTMGoogleAuthWorkaroundRelease` flag fix, as discussed on this Github issue: https://github.com/brave/brave-browser/issues/49300

However, I encountered a similar issue on another user's Linux desktop today. Turns out, that desktop had the `$XDG_CONFIG_HOME` shell environment variable set to a non-standard setting:

- Default XDG_CONFIG_HOME value: none (defaults to `$HOME/.config` when unset)

- Problematic user setting: `XDG_CONFIG_HOME=$HOME/config` (note that they set it to `config`, not `.config`)

Removing that XDG_CONFIG_HOME setting from their shell profile and rebooting solved the issue. May be of use to others.

**Comment by DwayneHawkins** [comment]
How do you know if it was resolved?

This happens like ~4 times per year for me, completely at random. So every couple of months this issue arrises.

I'm using endeavourOS and this is the only problem I have with Brave on Linux...

**Comment by wrobelda** [comment]
This happened to me two times this month and I think it's because my Gnome kills Brave when I log-out of my session. Each time I log back in I get a message it wasn't closed gracefully. As such I have the suspicion that it is the unclean closing that triggers this bug.

**Comment by LayersOfAbstraction** [comment]
> This happened to me two times this month and I think it's because my Gnome kills Brave when I log-out of my session. Each time I log back in I get a message it wasn't closed gracefully. As such I have the suspicion that it is the unclean closing that triggers this bug.

### Description

Only twice a month? You are a very lucky man. Happening **every day** to me on Pop!_OS 24.04 but **NOT** happening to me on Firefox.  I notice when I open the Brave Password manager that some of the entries have disappeared that should be there.


**TL;DR @wrobelda ** I haven't cloned the repo to check for any code that disposes of all Brave related resources during closing. If they didn't then implement that then this is going to be hard to fix if what you say is true as they would have to test this as the computer is shutting down unless they can do it in a container. 

### Steps to reproduce

- Power cycle a desktop computer.
- Start Brave.
- Log in to Facebook, LinkedIn and a few others.
- Ahh, I went to brave://settings/security and notice Secure DNS was on.  

### Actual result

Some of the  Password Manager entries in my main profile is gone. I'm noticing especially Facebook and LinkedIn.

### Expected result

It should work because I have followed the steps in [Martin Brinkmann's guide](https://chipp.in/guides/brave-signing-you-out-all-the-time-here-is-a-fix-for-that/)
 
1. Delete Browsing data on exit NOT enabled for anything. Everything is un-ticked.  

<img width="509" height="527" alt="Image" src="https://github.com/user-attachments/assets/360f04b3-e360-45d7-9dfa-0107ad2a060e" />

2. “Forget me when I close that site” is **not** checked.

<img width="385" height="668" alt="Image" src="https://github.com/user-attachments/assets/b8d3a534-0092-402e-a52d-15f67f20c491" />

### Reproduces how often

Shows up after usually 24 hours.  

### Channel information

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Brave version (brave://version info) 1.93.137 (Official Build) (x86_64)](https://brave.com/latest/)

### Miscellaneous information

Linux pop-os 7.0.11-76070011-generic #202606011647~1784215097~24.04~4974880 SMP PREEMPT_DYNAMIC Thu J x86_64 x86_64 x86_64 GNU/Linux

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ x] with Brave Rewards disabled
- [ x] in the latest version of Chrome


### TO DO: 

- [ ]  Copy all bookmarks to new profile. 
- [ ]  Start in a new profile with no add ons.
- [ ]  Save passwords to browser password manager.
- [ ] Still getting issues after 24 August?
